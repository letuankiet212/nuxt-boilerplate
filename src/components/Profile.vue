<template>
  <div class="card">
    <div class="card-content">
      <div class="media">
        <div class="media-left">
          <figure class="image is-128x128">
            <img :src="profile.imageUrl" alt="Placeholder image" />
          </figure>
        </div>
        <div class="media-content">
          <p class="title is-4">
            {{ profile.name }}
          </p>
          <p class="subtitle is-6">
            @
            {{ profile.id }}
          </p>
        </div>
      </div>
      <div class="content">
        {{ profile.description }}
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  props: {
    qiitaProfile: {
      type: Object,
      required: true,
      default() {
        return {
          id: 'qiita',
          name: 'Qiitaキータ',
          imageUrl:
            'https://qiita-image-store.s3.amazonaws.com/0/88/profile-images/1512392618',
          description:
            'Qiita公式アカウントです。Qiitaに関するお問い合わせに反応したり、お知らせなどを発信しています。'
        };
      }
    }
  },
  data() {
    return {
      profile: this.qiitaProfile
    };
  }
});
</script>
