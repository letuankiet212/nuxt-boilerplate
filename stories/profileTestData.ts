export default {
  id: 'qiita',
  name: 'Qiitaキータ',
  imageUrl:
    'https://qiita-image-store.s3.amazonaws.com/0/88/profile-images/1512392618',
  description:
    'Qiita公式アカウントです。Qiitaに関するお問い合わせに反応したり、お知らせなどを発信しています。'
};
