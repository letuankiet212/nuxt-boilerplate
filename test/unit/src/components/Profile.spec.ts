import { shallowMount } from '@vue/test-utils';
import Profile from '../../../../src/components/Profile.vue';
import profileTestData from './profileTestData';

describe('components/Profile', () => {
  it('should show the default user', () => {
    const wrapper = shallowMount(Profile);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should show the test user', () => {
    const wrapper = shallowMount(Profile, {
      propsData: { qiitaProfile: profileTestData }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
