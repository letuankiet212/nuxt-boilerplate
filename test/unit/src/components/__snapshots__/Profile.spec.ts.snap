// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Profile should show the default user 1`] = `
"<div class=\\"card\\">
  <div class=\\"card-content\\">
    <div class=\\"media\\">
      <div class=\\"media-left\\">
        <figure class=\\"image is-128x128\\"><img src=\\"https://qiita-image-store.s3.amazonaws.com/0/88/profile-images/1512392618\\" alt=\\"Placeholder image\\" data-test=\\"profile-image-url\\"></figure>
      </div>
      <div class=\\"media-content\\">
        <p class=\\"title is-4\\">
          Qiitaキータ
        </p>
        <p class=\\"subtitle is-6\\">
          @
          qiita
        </p>
      </div>
    </div>
    <div class=\\"content\\">
      Qiita公式アカウントです。Qiitaに関するお問い合わせに反応したり、お知らせなどを発信しています。
    </div>
  </div>
</div>"
`;

exports[`components/Profile should show the test user 1`] = `
"<div class=\\"card\\">
  <div class=\\"card-content\\">
    <div class=\\"media\\">
      <div class=\\"media-left\\">
        <figure class=\\"image is-128x128\\"><img src=\\"https://qiita-image-store.s3.amazonaws.com/0/71899/profile-images/1473698980\\" alt=\\"Placeholder image\\" data-test=\\"profile-image-url\\"></figure>
      </div>
      <div class=\\"media-content\\">
        <p class=\\"title is-4\\">
          keita koga
        </p>
        <p class=\\"subtitle is-6\\">
          @
          keitakn
        </p>
      </div>
    </div>
    <div class=\\"content\\">
      よろしくお願いします。
    </div>
  </div>
</div>"
`;
